compiler/hash-parser/src/parser/pat.rs

```diff
@@ -112,9 +108,9 @@ impl<'s> AstGen<'s> {
                 }
                 // An access pattern which accesses the `subject` with a particular `property`
                 // denotes with a name.
-                TokenKind::Colon if matches!(self.peek_second(), Some(token) if token.has_kind(TokenKind::Colon)) =>
+                TokenKind::Access =>
                 {
-                    self.frame.skip(2);
+                    self.skip_fast(); // `::`
```

Comment:
Would it make sense to have some kind of `self.skip_fast(TokenKind::Access)` just for a sanity check?

Reply:
I guess it can have a `debug_assert!()` in `skip_fast()`?

Reply:
Although this might be annoying to do when we skip literals since you can't just pass a pattern.

Reply:
Could use a macro for that: `debug_assert!(matches!(x, $token_pat))`. But anyway I don't know if this is worth it or not